external/elasticsearch/src/main/java/com/digitalpebble/stormcrawler/elasticsearch/persistence/AbstractSpout.java

```diff
@@ -164,13 +164,19 @@ public void open(
     protected final boolean addHitToBuffer(SearchHit hit) {
         Map<String, Object> keyValues = hit.getSourceAsMap();
         String url = (String) keyValues.get("url");
+
+        Map<String, List<String>> metadatas = (Map<String, List<String>>) keyValues.get("metadata");
+        addHitInfoToMetadata(metadatas, hit);
+
         // is already being processed - skip it!
         if (beingProcessed.containsKey(url)) {
```

Comment:
Move that right after the  line 166 

  _String url = (String) keyValues.get("url");_